docs: :memo: explain why we call it "check", not "validate" (#196)
seedcase-project/check-datapackage: docs/design/architecture.qmd

concepts specific to `check-datapackage`. The main objects and actions
used throughout the package can be found in the tables below.

> ### "Check" (vs "validate")

Reviewer: Or maybe even include "verify" so all three are mentioned in the header
```suggestion
### "Why "check" (and not "validate" or "verify")?
```

concepts specific to `check-datapackage`. The main objects and actions
used throughout the package can be found in the tables below.

### "Check" (vs "validate")

If you have ever searched for tools that check something against a
specification, you'll often see the word "validate". And you might
notice that we don't use the word "validate" in our package and
documentation. This is intentional.

Although the word is ubiquitous in programming, it's often used loosely
and in ways that deviate from its actual meaning. Tools that describe
themselves as "validating" something often, in practice, *verify* that
something matches a defined expectation or specification. There are many
websites and articles comparing the difference between validate and
verify. These two verbs often are done during quality control or

> assurance stages of building products and after are simplified to "V&V".

Author: ```suggestion
assurance stages of building products.
```

concepts specific to `check-datapackage`. The main objects and actions
used throughout the package can be found in the tables below.

### Why "check" and not "validate"?

If you have ever searched for tools that check something against a
specification, you'll often see the word "validate". And you might
notice that we never use the word "validate" in our package and
documentation. This is intentional.

While the word is ubiquitous in programming, it often doesn't actually

> mean what these tools do. What these tools often do is *verify*,

Reviewer: ```suggestion
Although the word is ubiquitous in programming, it's often used loosely and in ways that deviate
from its actual meaning. Tools that describe themselves as "validating" something often, in practice, *verify*,
```

*check*, or *confirm* that something matches some explicitly described
expectation or specification. There are many websites and articles
comparing the difference between validate and verify, and we won't get
into a detailed explanation here. For a good overview, see the Wikipedia

> on this topic in

Reviewer: ```suggestion
expectation or specification. There are many websites and articles
comparing the difference between validate and verify. For a good overview, see the Wikipedia
on this topic in
```
Removed this since "validate" is briefly described below.

[general](https://en.wikipedia.org/wiki/Verification_and_validation) and
on
[software](https://en.wikipedia.org/wiki/Software_verification_and_validation)
specifically.

Briefly, to "validate" something means to ask the question of "are we
building the right product". To "verify" something is asking the
question "are we building the product right". The key difference is
where the word "right" is placed.

> 

Author: Wait, you both are saying different things. @joelostblom is saying remove the one above, but @signekb you are saying to remove the one below.

: Actions that `check-datapackage` can perform.

### "Why "check" (and not "validate" or "verify")?

If you have ever searched for tools that check something against a
specification, you'll often see the word "validate". You might also
notice that we don't use the word "validate" in our package and
documentation. This is intentional.

Although the word "validate" is ubiquitous in programming, it's often

> used loosely and in ways that don't align from its actual meaning. Tools

Author: Verify is not as common. The whole point is that validate is used everywhere, even when it doesn't make sense.

that "validate" something often, in practice, *verify* that something
matches a defined expectation or specification. These two verbs are
often used during the quality control or assurance stages of building

> products. There are many websites and articles comparing the difference

Reviewer: ```suggestion
matches a defined expectation or specification. 
There are many websites and articles comparing the difference
```
